src/include/miopen/solver.hpp

```diff
@@ -979,6 +979,69 @@ struct PerformanceImplicitGemmXdlops : Serializable<PerformanceImplicitGemmXdlop
     std::string ToString() const;
 };
 
+struct PerformanceImplicitGemmForwardV4R4Xdlops
+    : Serializable<PerformanceImplicitGemmForwardV4R4Xdlops>
+{
+    int GemmMPerBlock;
+    int GemmNPerBlock;
+    int GemmKPerBlock;
+    int GemmMPerWave;
+    int GemmNPerWave;
+    int GemmG;
+    int GemmKPack;
+    bool GemmAThreadCopyMoreGemmK;
+    bool GemmBThreadCopyMoreGemmKPack;
+
+    bool use_spare_set;
+
+    PerformanceImplicitGemmForwardV4R4Xdlops(int, int, int, int, int, int, int, bool, bool, bool);
+
+    PerformanceImplicitGemmForwardV4R4Xdlops(
+        int a, int b, int c, int d, int e, int f, int g, bool h, bool i)
+        : PerformanceImplicitGemmForwardV4R4Xdlops(a, b, c, d, e, f, g, h, i, false)
+    {
+    }
+
+    PerformanceImplicitGemmForwardV4R4Xdlops()
+        : PerformanceImplicitGemmForwardV4R4Xdlops(-1, -1, -1, -1, -1, -1, -1, false, false, false)
+    {
+    }
+
+    PerformanceImplicitGemmForwardV4R4Xdlops(bool spare);
+
+    template <class Self, class F>
+    static void Visit(Self&& self, F f)
+    {
+        f(self.GemmMPerBlock, "GemmMPerBlock");
+        f(self.GemmNPerBlock, "GemmNPerBlock");
+        f(self.GemmKPerBlock, "GemmKPerBlock");
+        f(self.GemmMPerWave, "GemmMPerWave");
+        f(self.GemmNPerWave, "GemmNPerWave");
+        f(self.GemmG, "GemmG");
+        f(self.GemmKPack, "GemmKPack");
+        f(self.GemmAThreadCopyMoreGemmK, "GemmAThreadCopyMoreGemmK");
+        f(self.GemmBThreadCopyMoreGemmKPack, "GemmBThreadCopyMoreGemmKPack");
+    }
+
+    void EuristicInit(const ConvolutionContext& ctx);
+    bool IsValidValue() const;
+    bool SetNextValue();
+    bool IsValid(const ConvolutionContext& ctx) const;
+    bool operator==(const PerformanceImplicitGemmForwardV4R4Xdlops& other) const;
+    std::string ToString() const;
+
+    bool IsValidPerformanceConfig(const ConvolutionContext& ctx) const;
+    bool IsWorkAroundedPerformanceConfig(const ConvolutionContext& ctx) const;
+    bool IsFastPerformanceConfig(const ConvolutionContext& ctx) const;
```

Comment:
`IsFastToBeUsedForTuning()` ? It may look cumbersome or ugly, but it should be cumbersome and ugly because it reflects the cumbersome optimization logic.

Reply:
renamed